includes/rest/class-nodeinfo-controller.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ * NodeInfo controller file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Rest;
+
+use function Activitypub\get_total_users;
+use function Activitypub\get_active_users;
+use function Activitypub\get_rest_url_by_path;
+use function Activitypub\get_masked_wp_version;
+
+/**
+ * ActivityPub NodeInfo Controller.
+ *
+ * @author Matthias Pfefferle
+ *
+ * @see https://nodeinfo.diaspora.software/
+ */
+class Nodeinfo_Controller extends \WP_REST_Controller {
+	/**
+	 * The namespace of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $namespace = ACTIVITYPUB_REST_NAMESPACE;
+
+	/**
+	 * The REST base for this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'nodeinfo';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => '__return_true',
+				),
+			)
+		);
+
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<version>\d\.\d)',
+			array(
+				'args' => array(
+					'version' => array(
+						'description' => 'The version of the NodeInfo schema.',
+						'type'        => 'string',
+						'enum'        => array( '1.0', '1.1', '2.0', '2.1' ),
```

Comment:
any argument why not simply allow every version number like x.x? otherwise we would have to add every new version, if we support it or not!

(there is already a 2.2 btw. https://github.com/jhass/nodeinfo/tree/main/schemas/2.2 ☺️)

Reply:
Fixed in 2178c16.